build_test/lib/src/package_reader.dart

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:build/build.dart';
+import 'package:glob/glob.dart';
+import 'package:package_resolver/package_resolver.dart';
+
+/// Resolves using a [SyncPackageResolver] before reading from the file system.
+class PackageAssetReader implements AssetReader {
+  final SyncPackageResolver _packageResolver;
+
+  /// What package is the originating build occurring in.
+  final String _rootPackage;
+
+  /// Wrap a [SyncPackageResolver] to identify where files are located.
+  ///
+  /// To use a normal [PackageResolver] use `asSync`:
+  /// ```
+  /// new PackageAssetReader(await packageResolver.asSync);
+  /// ```
+  const PackageAssetReader(this._packageResolver, [this._rootPackage]);
+
+  /// A reader that can resolve files known to the current isolate.
+  static Future<PackageAssetReader> currentIsolate({String rootPackage}) async {
+    var resolver = PackageResolver.current;
+    return new PackageAssetReader(await resolver.asSync, rootPackage);
+  }
+
+  File _resolve(AssetId id) {
+    var nonLibPath = '${_packageResolver.packagePath(id.package)}/${id.path}';
```

Comment:
Now that this is simplified you could just use [p.canonincalize](https://www.dartdocs.org/documentation/path/1.4.1/path/canonicalize.html) to get a platform independent path